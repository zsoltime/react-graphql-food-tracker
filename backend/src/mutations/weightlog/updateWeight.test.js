'use strict';

const updateWeight = require('./updateWeight');

describe('updateWeight mutation', () => {
  it('updates the weight log fields', async () => {
    const args = {
      id: 'validFoodId',
      date: '2018-11-20',
      weight: 88.8,
    };
    const mockedContext = {
      db: {
        mutation: {
          updateWeightLog: jest.fn().mockImplementation(() => Promise.resolve({
            ...args,
            image: null,
          })),
        },
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([{ id: 'validUserId' }])),
        },
      },
      request: { userId: 'validUserId' },
    };

    const weight = await updateWeight(null, args, mockedContext);

    expect(mockedContext.db.query.users).toBeCalled();
    expect(
      mockedContext.db.query.users.mock.calls[0]
    ).toMatchSnapshot();
    expect(mockedContext.db.mutation.updateWeightLog).toBeCalled();
    expect(
      mockedContext.db.mutation.updateWeightLog.mock.calls[0]
    ).toMatchSnapshot();
    expect(weight).toMatchSnapshot();
  });

  it('throws error if user is not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      updateWeight(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
