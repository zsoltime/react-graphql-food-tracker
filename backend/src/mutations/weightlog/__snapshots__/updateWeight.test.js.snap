// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateWeight mutation throws error if user is not signed in 1`] = `"You must be signed in to update your weight"`;

exports[`updateWeight mutation updates the weight log fields 1`] = `
Array [
  Object {
    "where": Object {
      "AND": Array [
        Object {
          "id": "validUserId",
          "weightLogs_some": Object {
            "id": "validFoodId",
          },
        },
      ],
    },
  },
]
`;

exports[`updateWeight mutation updates the weight log fields 2`] = `
Array [
  Object {
    "data": Object {
      "date": "2018-11-20",
      "weight": 88.8,
    },
    "where": Object {
      "id": "validFoodId",
    },
  },
]
`;

exports[`updateWeight mutation updates the weight log fields 3`] = `
Object {
  "date": "2018-11-20",
  "id": "validFoodId",
  "image": null,
  "weight": 88.8,
}
`;
